test_data/series7_device_config.yaml

```diff
@@ -94,3 +94,54 @@ disabled_cell_bel_map:
      - TFF
      - IFF
      - OUTFF
+disabled_site_pips:
+  - bels:
+     - A6LUT
+     - B6LUT
+     - C6LUT
+     - D6LUT
+    ipin: A6
+    opin: O6
+bel_chains:
+- name: CARRY_CHAIN
+  source:
+    cell_port: CO[3]
+    bel_port: CO3
+  sink:
+    cell_port: CI
+    bel_port: CIN
+  sites:
+    - SLICEL
+    - SLICEM
+  cells:
+    - CARRY4
+  chain_drivers: # Place driver cell of S port to a proper BEL
+  - ports:
+    - name: S[0]
+      bels:
+        - AFF
```

Comment:
this seems like an interesting one, the `AFF` rule here is presumably just a convenient heuristic, there's no dedicated route.

care is also going to need to be taken in nextpnr around the case where different chain driver FFs in the same tile have different control sets so can't legally be placed (in which case the FF has to go to another site)

Reply:
Right, didn't think of the case in which there are different FFs feeding in the same carry. I guess that this case can still be handled by removing conflicting FFs from the clustering in nextpnr, therefore allowing the placer to freely chose another location